.github/workflows/main-build.yml

```diff
@@ -124,12 +134,22 @@ jobs:
           echo TERRAFORM_LAMBDA_FUNCTION_NAME=lambda-${{ matrix.language }}-${{ matrix.sample-app }}-${{ matrix.instrumentation-type }}-${{ matrix.architecture }}-${{ github.run_id }} |
           tee --append $GITHUB_ENV
       - name: Apply terraform
+        if: ${{ matrix.language != 'python' || matrix.python-version != '3.8' }}
```

Comment:
With above tweak, we wouldn't need this if.

Reply:
yeah, currently using this `if` condition to run on every matrix except for `python-3.8` and have this one trigger when meeting the [condition](https://github.com/aws-observability/aws-otel-lambda/pull/242/files#diff-cadecc0b07db7b001692f19a091a0fe88e550e6c04d8eb279431ea93bc6fe35eR145) . after few trail/error attempts found out this may be better way, which allows to have a seperate runs for both of them.

Reply:
Ideally we can keep the "weirdness" of the python 3.8 verification to only affect python, currently it is effecting all the workflows requiring several workarounds like the many `exclude`s above. I guess it's true we should probably have a separate run - in that case I think copying into a separate workflow only for python3.8 could make sure the normal workflow is still easy to reason about

Reply:
totally, It would be easy to have seperate workflow than to deal with few complexities involved in GH actions and to have all workflows affected. will modify this PR to have seperate workflow and leave `main-build.yml` uninterrupted, if that sounds like a better idea.